client/src/components/map.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useState, useEffect, useRef} from 'react';
+import Map from 'ol/Map';
+import View from 'ol/View';
+import Feature from 'ol/Feature';
+import Geometry from 'ol/geom/Geometry';
+import TileLayer from 'ol/layer/Tile';
+import VectorLayer from 'ol/layer/Vector';
+import VectorTileSource from 'ol/source/VectorTile';
+import VectorTileLayer from 'ol/layer/VectorTile';
+import MVT from 'ol/format/MVT';
+import VectorSource from 'ol/source/Vector';
+import XYZ from 'ol/source/XYZ';
+import {fromLonLat} from 'ol/proj';
+import * as styles from './map.module.scss';
+
+
+interface IMapWrapperProps {
+  features: Feature<Geometry>[],
+};
+
+// The below adapted from
+// https://taylor.callsen.me/using-openlayers-with-react-functional-components/
+const MapWrapper = ({features}: IMapWrapperProps) => {
+  const [map, setMap] = useState<Map>();
+  const [featuresLayer, setFeaturesLayer] = useState<VectorLayer>();
+
+  const mapElement = useRef() as
+        React.MutableRefObject<HTMLInputElement>;
+
+  useEffect( () => {
+    const initalFeaturesLayer = new VectorTileLayer({
+      source: new VectorTileSource({
+        format: new MVT(),
+        url: 'https://gis.data.census.gov/arcgis/rest/services/Hosted/VT_2019_150_00_PY_D1/VectorTileServer/tile/{z}/{y}/{x}.mvt',
+      }),
+    });
+
+    const initialMap = new Map({
+      target: mapElement.current,
+      layers: [
+
+        new TileLayer({
+          source: new XYZ({
+            url: 'https://{1-4}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}@2x.png',
+          }),
+        }),
+        initalFeaturesLayer,
+      ],
+      view: new View({
+        projection: 'EPSG:3857',
+        center: fromLonLat([-95.7129, 37.0902]), // [37.0902, 95.7129],
+        zoom: 3,
+      }),
+      controls: [],
+    });
+
+    setMap(initialMap);
+    setFeaturesLayer(initalFeaturesLayer);
```

Comment:
This line is giving me a typescript type warning: 
`
TS2345: Argument of type 'VectorTileLayer' is not assignable to parameter of type 'SetStateAction<VectorLayer | undefined>'.   Type 'VectorTileLayer' is not assignable to type 'VectorLayer'.     Property 'createRenderer' is protected in type 'VectorTileLayer' but public in type 'VectorLayer'
`